pyk/src/tests/integration/kllvm/test_prooftrace.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import cast
+
+import pyk.kllvm.load  # noqa: F401
+import pyk.kllvm.prooftrace as prooftrace
+from pyk.testing import ProofTraceTest
+
+
+class TestProofTrace(ProofTraceTest):
+    KOMPILE_DEFINITION = """
+        module TEST-PROOF-TRACE-SYNTAX
+          syntax Foo ::= a() | b() | c()
+        endmodule
+        
+        module TEST-PROOF-TRACE
+          imports TEST-PROOF-TRACE-SYNTAX
+          rule a() => b()
+          rule b() => c()
+        endmodule
+    """
+    KOMPILE_MAIN_MODULE = 'TEST-PROOF-TRACE'
+    HINTS_INPUT_KORE = r"""
+        LblinitGeneratedTopCell{}(
+          Lbl'Unds'Map'Unds'{}(
+            Lbl'Stop'Map{}(),
+            Lbl'UndsPipe'-'-GT-Unds'{}(
+               inj{SortKConfigVar{}, SortKItem{}}(\dv{SortKConfigVar{}}("$PGM")),
+               inj{SortFoo{}, SortKItem{}}(Lbla'LParRParUnds'TEST-PROOF-TRACE-SYNTAX'Unds'Foo{}())
+            )
+          )
+        )
+    """
```

Comment:
You could express this using the KORE DSL Pyk gives you rather than writing a string by hand.

Reply:
Thanks for the tip! Could you please point me to an example?

Reply:
Here's an integration test that constructs the expected output of a test using the KORE data structures; your input data here could be similar. Alternatively, you could just check the KORE text in as a test data file. Up to you what you prefer, but I think embedding KORE in a Python string is the least preferable of the three versions.

https://github.com/runtimeverification/k/blob/48b6ffa4110a30d5bd89f1952222c1d17d0d5dbe/pyk/src/tests/integration/kore/test_top_cell_initializer.py#L69-L78

Reply:
Thank you, Bruce! I could assemble this as a replacement:
```Python
    program_pattern = KoreParser("Lbla\'LParRParUnds\'TEST-PROOF-TRACE-SYNTAX\'Unds\'Foo{}()").pattern()

    HINTS_INPUT_KORE = init_generated_top_cell(
        map_pattern(
            (
                inj(SORT_K_CONFIG_VAR, SORT_K_ITEM, k_config_var('$PGM')),
                inj(SortApp('SortFoo'), SORT_K_ITEM, program_pattern),
            )
        )
    ).text
```

However, I would like very much to replace the argument of `KoreParser` with this `kore_text`
```Python
kore_text = _kast(definition_dir=x, input='program', output='kore', expression="a()").stdout
```

The issue with this approach is that I couldn't figure out how to get the `x` definition directory :/ 
Could you help with it? Or is it okay to let it as it is?

Reply:
Rather than _parsing_ that pattern, just construct an instance of `pyk.kore.syntax.App` directly: https://github.com/runtimeverification/k/blob/54d416a4dfcc9dc09e5a19774fb91f700ee6ea09/pyk/src/pyk/kore/syntax.py#L475

Reply:
The `pyk.kore.syntax` module defines data structures representing the KORE AST, the same as we have in the LLVM backend's AST module.

Reply:
Done! Could you please review this and the whole PR once more? Sorry about the trouble!